src/test/c_api_tests.cpp

```diff
@@ -40,3 +46,163 @@ TEST_F(CapiConfigTest, Parse) {
     EXPECT_EQ(ovms::Config::instance().restPort(), 234);
     EXPECT_EQ(ovms::Config::instance().configPath(), "/path/config.json");
 }
+
+class CapiInferencePreparationTest : public ::testing::Test {};
+
+TEST_F(CapiInferencePreparationTest, Basic) {
+    // request creation
+    OVMS_InferenceRequest* request{nullptr};
+    OVMS_Status* status = OVMS_InferenceRequestNew(&request, "dummy", 1);
+    ASSERT_EQ(nullptr, status);
+    ASSERT_NE(nullptr, request);
+
+    // adding input
+    status = OVMS_InferenceRequestAddInput(request, DUMMY_MODEL_INPUT_NAME, OVMS_DATATYPE_FP32, DUMMY_MODEL_SHAPE.data(), DUMMY_MODEL_SHAPE.size());
+    ASSERT_EQ(nullptr, status);
+    // setting buffer
+    std::array<float, DUMMY_MODEL_INPUT_SIZE> data{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    uint32_t notUsedNum = 0;
+    status = OVMS_InferenceRequestInputSetData(request, DUMMY_MODEL_INPUT_NAME, reinterpret_cast<void*>(data.data()), sizeof(float) * data.size(), OVMS_BUFFERTYPE_CPU, notUsedNum);
+    ASSERT_EQ(nullptr, status);
+    // add parameters
+    const uint64_t sequenceId{42};
+    status = OVMS_InferenceRequestAddParameter(request, "sequence_id", OVMS_DATATYPE_U64, reinterpret_cast<const void*>(&sequenceId), sizeof(sequenceId));
+    ASSERT_EQ(nullptr, status);
+    const uint32_t sequenceControl{1};  // SEQUENCE_START
+    status = OVMS_InferenceRequestAddParameter(request, "sequence_control_input", OVMS_DATATYPE_U32, reinterpret_cast<const void*>(&sequenceControl), sizeof(sequenceControl));
+    ASSERT_EQ(nullptr, status);
+
+    //////////////////
+    //  INFERENCE
+    //////////////////
+
+    ///////////////
+    // CLEANUP
+    ///////////////
+    // here we will add additional inputs to verify 2 ways of cleanup
+    // - direct call to remove whole request
+    // - separate calls to remove partial data
+    //
+    // here we will just add inputs to remove them later
+    // one original will be removed together with buffer during whole request removal
+    // one will be removed together with request but without buffer attached
+    // one will be removed with buffer directly
+    // one will be removed without buffer directly
+    status = OVMS_InferenceRequestAddInput(request, "INPUT_WITHOUT_BUFFER_REMOVED_WITH_REQUEST", OVMS_DATATYPE_FP32, DUMMY_MODEL_SHAPE.data(), DUMMY_MODEL_SHAPE.size());
```

Comment:
How do we ensure that this is removed? Do we only check if there is no segfault at the end of the test or?

Reply:
We do not ensure that it is removed. It is here only to show that you can (but not must) remove buffer/tensor independently from request.